armi/reactor/converters/axialExpansionChanger/axialExpansionChanger.py

```diff
@@ -342,12 +345,17 @@ def axiallyExpandAssembly(self):
                             # the top of the block below it
                             c.zbottom = self.linked.linkedBlocks[b][0].p.ztop
                     c.ztop = c.zbottom + c.height
-                    # update component number densities
+                    # update component number densities and detailed number densities
                     newNumberDensities = {
                         nuc: c.getNumberDensity(nuc) / growFrac
                         for nuc in c.getNuclides()
                     }
                     c.setNumberDensities(newNumberDensities)
```

Comment:
I tested this locally on the axial expansion unit tests in ARMI and we can simplify lines 349-353 with a call to `changeNDensByFactor(1.0 / growFrac)`. 

If you wrap the `if self.p.detailedNDens: ... ` changes into `changeNDensByFactor` then it'll be captured here too. That would be a nice simplification.

Reply:
This has been incorporated now.